clients/src/test/java/org/apache/kafka/common/record/FileLogInputStreamTest.java

```diff
@@ -209,6 +209,14 @@ public void testBatchIterationIncompleteBatch() throws IOException {
         }
     }
 
+    @Test
+    public void testNextBatchSelectionAtMaxIntValue() throws IOException {
```

Comment:
Maybe worth a test case which ensures correct behavior with start and end = 0 as well? Another good one would be to have a single batch which is truncated in the middle.

Reply:
This would be a good idea, except that if the case includes `position` equal to `size` we will always get `null`. The case I covered was about type overflow when both parameters were set to their maximum values, so I think about renaming this method instead.